src/main/webapp/js/calendar.js

```diff
@@ -167,7 +175,25 @@ function createSchedule(course, startDate, endDate) {
     isReadOnly: true,
   }]);
   // Each schedule needs a unique ID. We'll start at 1 and go up from there.
-  id++;
+  if (scheduleInfo[course.course_id] == null) {
```

Comment:
javascript has a weird concept of null vs undefined. my console actually shows `undefined` as the value that this would return. a safer was is just `course.course_id in scheduleInfo`